### PR TITLE
os-prober: init at 1.65

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change
Make os-prober available on nixos, and eventually solve https://github.com/NixOS/nixpkgs/issues/7406
With this package, I can use os-prober this way : 
```
  boot.loader.grub = {
    enable = true;
    version = 2;
    device = "/dev/sda";
    extraPrepareConfig = "PATH=$PATH:${pkgs.os-prober}/bin:${pkgs.busybox}/bin pkgdatadir=${pkgs.grub2}/share/grub ${pkgs.grub2}/etc/grub.d/30_os-prober > /boot/grub/grub-os-prober.cfg";
    extraEntries = ''
      menuentry 'Other OS' {
        configfile $prefix/grub-os-prober.cfg
      }
    '';
  };
```
Still not perfect, but at least it works.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/os-prober/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ stdenv.mkDerivation rec {
   version = "1.65";
   name = "os-prober-${version}";
   src = fetchurl {
-    url = "mirror://debian/debian/pool/main/o/os-prober_${version}.tar.xz";
+    url = "mirror://debian/pool/main/o/os-prober/os-prober_${version}.tar.xz";
     md5 = "a7e833555f54387a4798ffea8c2bf0d4";
```
Comment (Mic92) [inline pkgs/tools/misc/os-prober/default.nix]: please use sha256 instead of md5 as checksum.

Comment (symphorien) [inline pkgs/tools/misc/os-prober/default.nix]: Done

File: pkgs/tools/misc/os-prober/default.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          )
+        }" \
+        --run "[ -d /var/lib/os-prober ] || mkdir /var/lib/os-prober"
+    done;
+  '';
+
+  meta = {
+    description = "Utility to detect other OSs on a set of drives";
+    homepage = http://packages.debian.org/source/sid/os-prober;
+    license = stdenv.lib.licenses.gpl2Plus;
```
Comment (Mic92) [inline pkgs/tools/misc/os-prober/default.nix]: Do you want to maintain this package?

**Comment by Mic92** [comment]
I do not use grub myself, but this sound likes a useful addition to the existing grub module.

File: pkgs/tools/misc/os-prober/default.nix (hunk 2 above)
Comment (symphorien) [inline pkgs/tools/misc/os-prober/default.nix]: No.

**Comment by Mic92** [comment]
Oh. I noticed too late, that you opened the pull request against release. Please open it against master again.

**Comment by symphorien** [comment]
Ah, sorry. I reopened a new PR on master since it doesn't seem possible to reopen this one.

**Pull request merged**